rllib/policy/view_requirement.py

```diff
@@ -56,3 +60,47 @@ def __init__(self,
             float("-inf"), float("inf"), shape=())
         self.shift = shift
         self.used_for_training = used_for_training
+
+
+def get_default_view_requirements(policy: "Policy") -> ViewReqDict:
+    """Returns a default ViewRequirements dict (for backward compatibility).
+
+    Args:
+        policy (Policy): The Policy, for which to generate the ViewRequirements
+            dict.
+
+    Returns:
+        ViewReqDict: The default view requirements dict.
+    """
+    from ray.rllib.evaluation.postprocessing import Postprocessing
+    from ray.rllib.policy.sample_batch import SampleBatch
+
+    # Default view requirements (equal to those that we would use before
```

Comment:
Doesn't this have to be inferred from the sample batch? I.e., if my custom policy is generating some field with "custom1" name, it won't be in this list.

How about we instead put all sample fields in the train batch by default if no view requirements are present?

Reply:
I'm wondering if it will actually be less confusing to users to always do that. Basically, for every field, by default make it available in the training batch view, and then as an optimization allow algorithms to remove specific fields (i.e., IMPALA doesn't need NEXT_OBS).

This would also make the transition here simpler--- we shouldn't need to change the code of any of the existing policies (since the API is backwards compatible).